hw5-maybe: done

AntonYermilov opened the pull request.

AntonYermilov committed:
hw5-maybe: done

jzuken commented on hw5-maybe/.idea/compiler.xml:
```diff
@@ -0,0 +1,9 @@
+<project version="4">
+  <component name="CompilerConfiguration">
+    <bytecodeTargetLevel>
+      <module name="hw5-maybe_main" target="1.8" />
```
это лучше указывать в конфиге gradle

jzuken commented on hw5-maybe/.idea/workspace.xml:
```diff
@@ -0,0 +1,458 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
а вот это выкладывать не нужно. а всё потому, что в .gitignore написано искать эти файлы только в корне проекта
-1

jzuken commented on hw5-maybe/src/main/java/me/eranik/NumberReader.java:
```diff
@@ -0,0 +1,49 @@
+    /**
+     * Reads numbers from the input file and writes their squares to the output file.
+     * @param args names of input and output files
+     * @throws IOException
```
исключения тоже документируем

jzuken commented on hw5-maybe/src/main/java/me/eranik/utils/Maybe.java:
```diff
@@ -0,0 +1,81 @@
+     * @param <U> type of Maybe
+     * @return new Maybe with specified value
+     */
+    public static <U> Maybe<U> just(U value) {
```
а вот тут аннотации NotNull/Nullable будут очень кстати

jzuken commented on hw5-maybe/src/main/java/me/eranik/utils/Maybe.java:
```diff
@@ -0,0 +1,81 @@
+
+import java.util.function.Function;
+
+public class Maybe<T> {
```
доки?
-0.5

jzuken commented on hw5-maybe/src/main/java/me/eranik/utils/Maybe.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    /**
+     * Returns value that is stored in Maybe. Throws exception if value does not exist.
```
кусок про исключение надо перенести на две строчки ниже

jzuken commented on hw5-maybe/src/main/java/me/eranik/utils/Maybe.java:
```diff
@@ -0,0 +1,81 @@
+     */
+    public T get() throws MaybeException {
+        if (value == null)
+            throw new MaybeException();
```
и что оно означает? исключения обычно именуют так, чтобы по имени класса уже понятная была проблема

jzuken commented on hw5-maybe/src/main/java/me/eranik/utils/Maybe.java:
```diff
@@ -0,0 +1,81 @@
+     */
+    public <U> Maybe<U> map(Function<? super T, ? extends U> mapper) {
+        if (value == null)
+            return new Maybe<>();
```
nothing()?

jzuken commented on hw5-maybe/src/main/java/me/eranik/NumberReader.java:
```diff
@@ -0,0 +1,49 @@
+        Scanner input = new Scanner(new InputStreamReader(new FileInputStream(inputFile)));
+        PrintWriter output = new PrintWriter(new OutputStreamWriter(new FileOutputStream(outputFile)));
+
+        while (input.hasNext()) {
```
это хочется тоже в виде теста

AntonYermilov committed:
hw5-maybe:
MaybeException renamed to AccessToNothingException.
Added description to the exceptions.
Added source compatibility to gradle.
Added tests for NumberReader.

AntonYermilov committed:
fixed patterns in .gitignore

jzuken commented on hw5-maybe/src/test/java/me/eranik/NumberReaderTest.java:
```diff
@@ -0,0 +1,84 @@
+    @BeforeEach
+    void createFiles() throws IOException {
+        input = Paths.get("src", "test", "resources", "input.txt").toFile();
+        input.createNewFile();
```
а папки resources нет, и никто её не создал тут.
https://stackoverflow.com/questions/6666303/javas-createnewfile-will-it-also-create-directories
тесты валятся как в консоли, так и в идее.
-1

AntonYermilov committed:
hw5-maybe: NumberReaderTest now creates resources dir

AntonYermilov committed:
hw5-maybe: added gradle-wrapper

jzuken committed:
Merge branch 'master' into hw5-maybe
